homeassistant/components/roon/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for roon integration."""
+import logging
+
+from roon import RoonApi
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+
+from .const import DEFAULT_NAME, DOMAIN, ROON_APPINFO  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str})
+
+
+class RoonHub:
+    """Interact with roon during config flow."""
+
+    def __init__(self, host):
+        """Initialize."""
+        self._host = host
+
+    async def authenticate(self) -> bool:
+        """Test if we can authenticate with the host."""
+        try:
+            roonapi = RoonApi(ROON_APPINFO, None, self._host)
```

Comment:
Make sure to run any blocking I/O in our executor thread pool with `hass.async_add_executor_job`. If the method `authenticate` does more than one call to blocking I/O it's better to make the whole method sync, and schedule it on the executor.

Reply:
This gave me some issues with json serialisation -but in any case the previous `config_flow` code had a non blocking approach (which is probably better because it does a better job of timing out).

So I've reused the non blocking logic of the previous code.